templates/admin/amp-status.php

```diff
@@ -19,15 +19,25 @@
 		<span class="screen-reader-text"><?php esc_html_e( 'Edit Status', 'amp' ); ?></span>
 	</a>
 	<div id="amp-status-select" class="hide-if-js" data-amp-status="<?php echo esc_attr( $status ); ?>">
-		<input id="amp-status-enabled" type="radio" name="<?php echo esc_attr( self::STATUS_INPUT_NAME ); ?>" value="enabled" <?php checked( ! $disabled ); ?>>
-		<label for="amp-status-enabled" class="selectit"><?php echo esc_html( $labels['enabled'] ); ?></label>
-		<br />
-		<input id="amp-status-disabled" type="radio" name="<?php echo esc_attr( self::STATUS_INPUT_NAME ); ?>" value="disabled" <?php checked( $disabled ); ?>>
-		<label for="amp-status-disabled" class="selectit"><?php echo esc_html( $labels['disabled'] ); ?></label>
-		<br />
-		<?php wp_nonce_field( self::NONCE_ACTION, self::NONCE_NAME ); ?>
+		<?php if ( $available ) : ?>
+			<fieldset>
+				<input id="amp-status-enabled" type="radio" name="<?php echo esc_attr( self::STATUS_INPUT_NAME ); ?>" value="enabled" <?php checked( ! $disabled ); ?>>
+				<label for="amp-status-enabled" class="selectit"><?php echo esc_html( $labels['enabled'] ); ?></label>
+				<br />
+				<input id="amp-status-disabled" type="radio" name="<?php echo esc_attr( self::STATUS_INPUT_NAME ); ?>" value="disabled" <?php checked( $disabled ); ?>>
+				<label for="amp-status-disabled" class="selectit"><?php echo esc_html( $labels['disabled'] ); ?></label>
+				<br />
+				<?php wp_nonce_field( self::NONCE_ACTION, self::NONCE_NAME ); ?>
+			</fieldset>
+		<?php else : ?>
+			<div class="inline notice notice-warning">
+				<p><?php esc_html_e( 'AMP cannot be enabled on home page, front page, password protected posts and post types which do not support AMP.', 'amp' ); ?></p>
```

Comment:
I think this message should only be shown if the post is the homepage, page for posts, password protected posts, or posts that don't support AMP. In other cases, we should show a different message such as when a plugin blocks AMP for a post via the `amp_skip_post` filter. For example:

> A plugin or theme has disabled AMP support.

Reply:
Also, the translation string should be changed to replace “front page” with “page for posts”. You can also use “homepage” instead of having a space in there. 

Per core:
![image](https://user-images.githubusercontent.com/134745/33778559-1416655c-dbfe-11e7-8fe6-d522345aa120.png)

Reply:
Also probably should say that AMP cannot _yet_ be enabled for those certain pages. Our intention is to eventually allow every page to be AMPed.

Reply:
Thanks for your feedback @westonruter. I improved the error messages to be more accurate in [this commit](https://github.com/Automattic/amp-wp/pull/825/commits/269b937c27a89cb9e42501b2b0706ae8654623a5). I had to add `amp_post_supports_error()` function to avoid redoing all checks.

<img width="276" alt="screen shot 2017-12-08 at 8 16 00 pm" src="https://user-images.githubusercontent.com/3581402/33781413-d538137e-dc54-11e7-8d5a-bf3e1af68bce.png">
<img width="276" alt="screen shot 2017-12-08 at 8 16 20 pm" src="https://user-images.githubusercontent.com/3581402/33781414-d553d910-dc54-11e7-993a-38eae61ca096.png">
<img width="274" alt="screen shot 2017-12-08 at 8 16 34 pm" src="https://user-images.githubusercontent.com/3581402/33781415-d56e9994-dc54-11e7-8f38-210dd025c21a.png">
<img width="272" alt="screen shot 2017-12-08 at 8 16 59 pm" src="https://user-images.githubusercontent.com/3581402/33781416-d58d2936-dc54-11e7-9e4b-8cdd8c78abc1.png">